src/vm/exceptionhandling.cpp

```diff
@@ -1814,7 +1814,25 @@ CLRUnwindStatus ExceptionTracker::ProcessOSExceptionNotification(
 
         if (fIsFrameLess)
         {
-            pGSCookie = (GSCookie*)cfThisFrame.GetCodeManager()->GetGSCookieAddr(cfThisFrame.pRD,
+            PREGDISPLAY pRD = NULL;
```

Comment:
@janvorli ``ExceptionTracker::InitializeCrawlFrame`` (called at the beginning of ``ExceptionTracker::ProcessOSExceptionNotification``) initializes cfThisFrame.pRD via `` Thread::InitRegDisplay`` call.

``Thread::InitRegDisplay`, however, does not initialize pRD, and thus #8981 attempts to initialize  pEbp inside  `` Thread::InitRegDisplay``.

At that point, the issue was that the context that ``Thread::InitRegDisplay`` takes is not that of the current frame, but that of the caller's frame. (That is why the changes on ``dispatcherContext.ContextRecord`` is required in #8981).

I currently have no overview of x86/Windows unwider, and thus I'm not sure whether it is possible to eliminate pXXX fields without leaving any side-effect on x86/Windows.

Anyway, I'll try that approach.

I have a question related with this. ``EECodeManager::GetGSCookieAddr`` has the following code:
```
    if  (info->ebpFrame)
    {
        return PVOID(SIZE_T((DWORD(*pContext->pEbp) - info->gsCookieOffset)));
    }
```

Could you let me know the purpose of ``pEbp``? I thought that it will point to the location where the previous frame pointer is stored, but the above code seems to assume that it will point to the current frame pointer.